enhancements/ingress/haproxy-log-format.md

```diff
@@ -0,0 +1,272 @@
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+Since MicroShift does not run the cluster-ingress-operator, there is
+no impact to MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+HAProxy can forward access logs to a syslog endpoint, and openshift-router
+recognizes environment variables to enable this feature and specify
+the log formats. Currently the only log format is HttpLogFormat
+(`ROUTER_SYSLOG_FORMAT`). Additional environment variables for
+HttpsLogFormat (`ROUTER_HTTPS_LOG_FORMAT`) and TcpLogFormat
+(`ROUTER_TCP_LOG_FORMAT`) will be recognized by the openshift-router
+to configure custom log formats for HTTPS and TCP logs, respectively.
+
+Note that the current environment variable for HttpLogFormat
+(`ROUTER_SYSLOG_FORMAT`) is non-specific and should be renamed to
+(`ROUTER_HTTP_LOG_FORMAT`). The old environment variable
+(`ROUTER_SYSLOG_FORMAT`) should still be honored, but the new
+environment variable (`ROUTER_HTTP_LOG_FORMAT`) will take precedence.
+
+### Risks and Mitigations
+
+The meaning of the specified log formats depends on the underlying
+implementation.  If we were to change away from HAProxy, the semantics of the
+format would likely change.  We can mitigate this risk by documenting
+it.
+
+### Drawbacks
+
+N/A
+
+## Alternatives (Not Implemented)
+
+It is possible to configure both `httpslog` and `tcplog` formats by
+directly adding those options to the HAProxy configuration
+template. While this would be an improvement, it would not allow for
+custom log formats as can currently be done for `HttpLogFormat`. In
+contrast, this proposal aims to provide a method to define custom log
+formats for `HttpsLogFormat` and `TcpLogFormat` that are consistent
+with the existing `HttpLogFormat`.
+
+## Open Questions [optional]
+
+N/A
+
+## Test Plan
+
+There are no planned e2e or integration tests for this
+enhancement. This is largely due to the fact that log format is a
+free-form string.
+
+## Graduation Criteria
+
+N/A
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+N/A
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+An upgrade/downgrade of the IngressOperator will result is the router
+image also being upgraded/downgraded. As such, openshift-router will
+recognize the new variables for log formats.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+## Support Procedures
+
+If any format string contains an invalid variable, HAProxy will exit
+immediately without providing a descriptive log message. For example, if a user includes a unrecognized format alias (eg. %GG), haproxy will emit the following error message:
+
+```
+[ALERT]    (168045) : config : Parsing [/etc/haproxy/haproxy.cfg:34]: failed to parse log-format : no such format alias 'GG'. If you wanted to emit the '%' character verbatim, you need to use '%%'.
+[ALERT]    (168045) : config : Fatal errors found in configuration.
```

Comment:
In my opinion, this fatal error is not acceptable for a new feature.  See https://github.com/openshift/enhancements/pull/1832/files#r2338077374